packages/theme-check-common/src/checks/valid-settings-key/index.spec.ts

```diff
@@ -0,0 +1,317 @@
+import { expect, describe, it } from 'vitest';
+import { ValidSettingsKey } from './index';
+import { check } from '../../test';
+
+describe('Moduel: ValidSettingsKey', () => {
+  const schemaTemplate = {
+    name: 'Example',
+    settings: [
+      {
+        id: 'example-text-setting',
+        type: 'text',
+        label: 'Example Text Setting',
+      },
+    ],
+  };
+
+  describe('default settings', () => {
+    it('does not report an error when default setting exists', async () => {
+      const theme = {
+        'sections/example.liquid': toLiquidFile({
+          ...schemaTemplate,
+          default: {
+            settings: {
+              'example-text-setting': 'value',
+            },
+          },
+        }),
+      };
+
+      const offenses = await check(theme, [ValidSettingsKey]);
+      expect(offenses).to.have.length(0);
+    });
+
+    it('reports an error when default setting does not exist', async () => {
+      const theme = {
+        'sections/example.liquid': toLiquidFile({
+          ...schemaTemplate,
+          default: {
+            settings: {
+              'non-existent-setting': 'value',
+            },
+          },
+        }),
+      };
+
+      const offenses = await check(theme, [ValidSettingsKey]);
+      expect(offenses).to.have.length(1);
+      expect(offenses[0].message).to.equal(
+        `Setting 'non-existent-setting' does not exist in schema.`,
+      );
+    });
+
+    it('does not report an error when default settings are defined in a block schema', async () => {
```

Comment:
Took me a second to see what we were doing here, maybe we could clarify this


```suggestion
    it('does not report an error when missing default settings is referenced from a block schema', async () => {
```